test/pkg/environment/environment.go

```diff
@@ -0,0 +1,86 @@
+package environment
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"knative.dev/pkg/environment"
+	loggingtesting "knative.dev/pkg/logging/testing"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/karpenter/pkg/apis"
+	"github.com/aws/karpenter/pkg/utils/env"
+)
+
+var EnvironmentName = flag.String("environment-name", env.WithDefaultString("ENVIRONMENT_NAME", ""), "Environment name enables discovery of the testing environment")
+
+type Environment struct {
+	context.Context
+	gomega.Gomega
+	Options *Options
+	Client  client.Client
+}
+
+func NewEnvironment(t *testing.T) (*Environment, error) {
+	ctx := loggingtesting.TestContextWithLogger(t)
+	client, err := NewLocalClient()
+	if err != nil {
+		return nil, err
+	}
+	options, err := NewOptions()
+	if err != nil {
+		return nil, err
+	}
+	g := gomega.NewGomega(ginkgo.Fail)
+	g.SetDefaultEventuallyTimeout(10 * time.Minute)
```

Comment:
Nice! Didn't know this existed. Maybe do `g.SetDefaultEventuallyPollingInterval(1*time.Second)` as well, looks like the default is 10ms